perses/annihilation/new_relative.py

```diff
@@ -0,0 +1,1213 @@
+        angle_parameters : list
+            list of angle parameters
+        """
+        index_set = set(indices)
+
+        #now loop through and try to find the angle:
+        for angle_index in range(angle_force.getNumAngles()):
+            angle_parameters = angle_force.getAngleParameters(angle_index)
+
+            #get a set representing the angle indices
+            angle_parameter_indices = set(angle_parameters[:3])
+
+            if index_set==angle_parameter_indices:
+                return angle_parameters
+
+        raise ValueError("The provided force does not contain the angle of interest.")
+
+    def _find_torsion_parameters(self, torsion_force, indices):
+        """
+        Convenience function to find the torsion parameters corresponding to a particular set of indices.
+
+        Parameters
+        ----------
+        torsion_force : openmm.PeriodicTorsionForce
+            torsion force where the torsion of interest may be found
+        indices : list of int
+            The indices (any order) of the atoms of the torsion
+
+        Returns
+        -------
+        torsion_parameters : list
+            torsion parameters
+        """
+        index_set = set(indices)
+
+        #now loop through and try to find the torsion:
+        for torsion_index in range(torsion_force.getNumTorsions()):
+            torsion_parameters = torsion_force.getTorsionParameters(torsion_index)
+
+            #get a set representing the torsion indices:
+            torsion_parameter_indices = set(torsion_parameters[:4])
+
+            if index_set==torsion_parameter_indices:
+                return torsion_parameters
+
+        raise ValueError("The torsion of interest was not found.")
+
+    def handle_harmonic_angles(self):
+        """
+        This method adds the appropriate interaction for all angles in the hybrid system. The scheme used, as with bonds, is:
+
+        1) If the three atoms are all in the core, then we add to the CustomAngleForce and interpolate between the two
+            parameters
+        2) Otherwise, we add the angle to a regular angle force.
+        """
+        old_system_angle_force = self._old_system_forces['HarmonicAngleForce']
+        new_system_angle_force = self._new_system_forces['HarmonicAngleForce']
+
+        #first, loop through all the angles in the old system to determine what to do with them. We will only use the
+        #custom angle force if all atoms are part of "core." Otherwise, they are either unique to one system or never
+        #change.
+        for angle_index in range(old_system_angle_force.getNumAngles()):
+            angle_parameters = old_system_angle_force.getAngleParameters(angle_index)
+
+            #get the indices in the hybrid system
+            hybrid_index_set = {self._old_to_hybrid_map[old_index] for old_index in angle_parameters[:3]}
+            #make a list because python set objects can't be indexed
+            hybrid_index_list = list(hybrid_index_set)
```

Comment:
I may not understand the logic here, but I think that building the list this way may randomly scramble the order of the atom indices, which could mean you'd end up controlling another angle.